aten/src/ATen/native/sparse/ValidateCompressedIndicesCommon.h

```diff
@@ -241,52 +266,79 @@ void _validate_compressed_sparse_indices_kernel(
 
     const auto batch_dims = cidx.sizes().slice(0, cidx.dim() - 1);
     const auto batch_count = indexCount(batch_dims);
-    const auto batch_idx = at::arange(batch_count, cidx.options()).view(batch_dims).unsqueeze_(-1);
+    const auto batch_idx =
+        at::arange(batch_count, cidx.options()).view(batch_dims).unsqueeze_(-1);
+
+    const Tensor batch_shape = at::tensor(batch_dims, idx.options());
+    const Tensor batch_idx_stride =
+        at::tensor(idx.strides().slice(0, idx.dim() - 1), idx.options());
```

Comment:
Shape and strides are int64. Let us probably keep it that way for these newly created tensors.

Reply:
Resolved.